server/context.go

```diff
@@ -131,6 +134,39 @@ type Context struct {
 	TimeUntilStoreDead time.Duration
 }
 
+func getDefaultCacheSize() int64 {
+	mem := sigar.Mem{}
+	// Returns uint.
+	err := mem.Get()
+	if err != nil {
+		log.Infof("system available memory info is invalid, setting rocksdb cache size to 512MB")
+		return 512 << 20
+	}
+	sysAvlMem := int64(mem.Total)
+	if sysAvlMem < 0 {
+		log.Infof("system available memory info is weird, setting rocksdb cache size to 512MB")
+		return 512 << 20
+	}
+	if runtime.GOOS == "linux" {
+		buf, err := ioutil.ReadFile(defaultCGroupMemPath)
+		if err != nil {
+			log.Infof("can't read available memory from cgroups, setting rocksdb cache size to half of system memory")
+			return sysAvlMem / 2
+		}
+
+		cgAvlMem, err := strconv.ParseInt(string(buf), 10, 64)
+		if err != nil {
+			log.Infof("can't undestand format of %s, setting rocksdb cache size to half of system memory", defaultCGroupMemPath)
+			return sysAvlMem / 2
+		}
+		if sysAvlMem < cgAvlMem {
```

Comment:
if you change this to `if cgAvlMem < sysAvlMem { return cgAvlMem/2 }`, then you can drop line 165.